### ISSUE TITLE
error with load_from_mallet_state

### ISSUE THREAD

**Issue by benmiller314** [agoldst/dfrtopics#6]
First, thank you for sharing all this code and writing such excellent walkthroughs, both here and in dfr-browser! 

I have several mallet `topic-state.gz` files, generated before I started using your package, that I'm trying now to load with the package, in part to take advantage of `export_browser_data()`. But the `load_from_mallet_state()` function has been crashing around the point of saving a simple state.csv and loading it into memory. 

Here's the error message I get when I call `load_from_mallet_state(mstate, simplified_state=simple_state, instances_file=ilist)` with absolute paths for `mstate`, `simple_state`, and `ilist`. (I found that bash-style filepath expansions, relative to ~, weren't being interpreted properly.)

> ```
> Loading /{mypath}/{myfile}.csv to a big.matrix...
> Done.
> Error in bigtabulate(x, ccols = ccols, breaks = breaks, table = FALSE,  : 
>   REAL() can only be applied to a 'numeric', not a 'logical'
> ```

Any idea what could be causing the mis-type? I see that the simplify_state() function calls Python; the relevant version on my Mac OS system is 2.7.10. But I checked in Terminal that `simplify_state.py` produces reasonable-looking, integer-based output, and it does, so I'm thinking the problem must be somewhere further down the rabbit-hole that starts with `read_state_bigmem()`.

Any help would be greatly appreciated, and please let me know what other information from me might be useful!

**Comment by agoldst** [comment]
Thanks for the detailed report! I'd heard of this error from another user who was on Windows and couldn't reproduce it on my Mac, but if you're on a Mac then maybe we can pin it down. The error is issued from `bigtabulate`'s C++ code, I'd guess. Would you be willing to send the results of a `traceback()`? Or better yet would you be willing to share your `topic-state.gz` and InstancesList files that are yielding the error? first name dot last name at rutgers dot edu.

In the meantime, as a workaround, you can pass `bigmemory=F` to `load_from_mallet_state` (hopefully that will actually work as a workaround).

**Comment by benmiller314** [comment]
Thanks for the quick reply! The `traceback()` is below, since it probably should have been in my first post, and I'll send the files by email. 

```
5: .Call("BigMatrixTAPPLY", x@address, as.numeric(ccols), as.numeric(breakm), 
       table, table.useNA, summary, as.numeric(summary.cols), summary.na.rm, 
       splitcol, splitlist, PACKAGE = "bigtabulate")
4: bigtabulate(x, ccols = ccols, breaks = breaks, table = FALSE, 
       useNA = useNA, splitcol = splitcol, splitret = splitret)
3: bigtabulate::bigsplit(ss, c("doc", "topic"))
2: read_state_bigmem(simplified_state_file, K = length(hyper$alpha))
1: load_from_mallet_state(mstate, simplified_state_file = simple_state, 
       instances_file = ilist)
```

**Comment by agoldst** [comment]
Okay, I think this really is a bug in the latest version of bigtabulate (1.1.5), which fortunately can be worked around. The issue appears to be that you are supposed to be able to pass `splitcol=NA` to `bigsplit` to request a list of row indices corresponding to each level of the split, but the underlying C++ code has a `REAL(splitcol)` ([here](https://github.com/kaneplusplus/bigtabulate/blob/master/src/bigtabulate.cpp#L480)) and plain old `NA` secretly has mode logical; hence the type error. But we can just set `splitcol=NA_real_` to defeat the problem. I've made the corresponding change in dfrtopics (at 22355038ccafc60dc3992b6b73d0c0eb401b5b82). With luck, that will now work for you; please let me know whether it does! I'll try to construct a minimal example for the bigtabulate developers.

**Comment by agoldst** [comment]
Seems like this is fixed by 22355038ccafc60dc3992b6b73d0c0eb401b5b82, though I will keep an eye on bigtabulate updates.
